[beam] add DayDivider to list views

crabinak opened the pull request:
@agritheory 
I've added a DayDivider to the Beam ListView. I centered-aligned the text to help it visually stand out amongst the lists. The medium gray worked best here as the darker colors overpowered the navigation menu. This takes in month, day, and year as props and formats them to display, but it could be reworked to take in an object or however the date would be passed in.

crabinak committed:
added DayDivider to Beam

agritheory commented on examples/beam/default.story.vue:
```diff
@@ -49,6 +49,28 @@
+				<template #navbaraction>Done</template>
+			</Navbar>
+
+			<BeamDayDivider :month="'10'" :day="'20'" :year="'2024'" />
```
Something like this but perhaps with a pre-formatted date. 

```const itemsWithDayDivider = computed(() => {
	items.splice(5, 0, {
			label: new Date(),
			linkComponent: 'DayDivider',
		})
	return items
})```

agritheory reviewed (changes requested):
I'd like to see the specimen use the ListView component API, I left an example that works but  doesn't look right

![image](https://github.com/user-attachments/assets/cccb5ff4-72fb-49aa-bcda-68355e54cba1)

Commit:
Merge remote-tracking branch 'upstream/development' into feat-beam-listview-day-divider

Alchez commented:
@crabinak I've added a commit that merges changes from the main branch, so you will have to pull before making more changes.

Commit:
Merge remote-tracking branch 'upstream/development' into feat-beam-listview-day-divider

crabinak committed:
fixed some padding styles for the list elements and updated prefixed variables

crabinak committed:
DayDivider now uses linkComponent. Added sample to data.json.

crabinak committed:
updated daydivider prop to accept Date object. Added optional flag for ISOString output vs. dateString output

crabinak commented:
@agritheory I updated this to use linkComponent. It accepts a Date object as the prop as well as an option to specify the format. As an example, I provided two date dividers: one as the default DateString and the other using ISOString. I didn't add more format types just incase this feature was not desired. 

I also left-aligned the DayeDivider content as having it center-aligned made it stand out but not in a good way.

Commit:
Merge remote-tracking branch 'upstream/development' into feat-beam-listview-day-divider

Commit:
fix: modify date formatter

Alchez commented:
@agritheory @crabinak 

I added some minor processing changes, and here's what the latest iteration looks like:

![image](https://github.com/user-attachments/assets/5bd0826a-ec2c-4512-8b4a-2db6f48b7b55)

Let me know if that looks okay.

agritheory commented:
@crabinak @Alchez How do we feel about using the lighter blue "cell change color" as background for this?

crabinak committed:
updated day divider to use cell-changed color

crabinak commented:
@Alchez Looks good to me. 

@agritheory I updated this to use that light blue color you requested. I also removed the top border from the day divider as it was causing a double border with the bottom border from the previous item.
